src/vm/exceptionhandling.cpp

```diff
@@ -4587,10 +4587,10 @@ VOID DECLSPEC_NORETURN UnwindManagedExceptionPass1(PAL_SEHException& ex)
 
             *currentFlags = firstPassFlags;
 
-            // Pop the last managed frame so that when the native frames are unwound and
-            // the UnwindManagedExceptionPass1 is resumed at the next managed frame, that
-            // managed frame is the current one set in the thread object.
-            GetThread()->GetFrame()->Pop();
+            // Pop all frames that are below the targed managed frame so that when the native
```

Comment:
Actually, I can see that the comment is misleading. It should say:
            // Pop all frames that are below the block of native frames and that would be
            // in the unwound part of the stack when UnwindManagedExceptionPass1 is resumed 
            // at the next managed frame.